driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -106,6 +109,28 @@ public void prepareReceived(final ChannelHandlerContext ctx, MessageEvent evt) t
             return;
         }
 
+        if ((lastRobot != null && ! lastRobot.getDisposedFuture().isDone()) || ! testFinished.compareAndSet(true, false)) {
```

Comment:
Adding a synchronization between threads for sure is adding complexity. You cannot have without the other.

Regarding the queue, I guess you mean to keep the requests to be processed in a queue. I also need the dispose future for the current executing robot. I don't know how to have a Robot and a list of Runnable-s or PrepareEvent-s in one queue and then remove a Robot and a Runnable but add at the beginning of the queue another Robot. I don't even know what kind of structure can I use for this.

I tried also to think how to replace the atomic boolean with a queue. I don't know how to guarantee that a new test coming exactly when the previous one finishes will be executed. I still need some kind of synchronization which will add more complexity.

If you have any suggestion on how to do it, please let me know.